### PR TITLE
Parse step response when JSON schema

### PR DESCRIPTION
**Pull request by neoxelox**

### CODE REVIEW

File: src/compiler/base/nodes/tags/step.ts [hunk 1]
```diff
@@ -80,7 +71,12 @@ export async function compile(
     }
 
     const textVarValue = (stepResponse?.content ?? []).filter(c => c.type === ContentType.text).map(c => c.text).join('')
-    scope.set(String(textVarName), textVarValue)
+
+    if ("schema" in config) {
+      scope.set(String(textVarName), JSON.parse(textVarValue.trim()))
```
Comment (andresgutgon) [inline src/compiler/base/nodes/tags/step.ts:76]: Try / catch this. Can fail

Comment (neoxelox) [reply src/compiler/base/nodes/tags/step.ts:76]: We want to fail

Comment (neoxelox) [reply src/compiler/base/nodes/tags/step.ts:76]: But I could throw a promptl error instead of json error maybe

Comment (andresgutgon) [reply src/compiler/base/nodes/tags/step.ts:76]: ;)

**Pull request merged**